plugins/modules/cloud/opennebula/one_template.py

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/python
+#
+# Copyright: (c) 2021, Georg Gadinger <nilsding@nilsding.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: one_template
+
+short_description: Manages OpenNebula templates
+
+version_added: 2.4.0
+
+requirements:
+  - pyone
+
+description:
+  - "Manages OpenNebula templates."
+
+options:
+  id:
+    description:
+      - A I(id) of the template you would like to manage.  If not set then a
+      - new template will be created with the given I(name).
+    type: int
+  name:
+    description:
+      - A I(name) of the template you would like to manage.  If a template with
+      - the given name does not exist it will be created, otherwise it will be
+      - managed by this module.
+    type: str
+  template:
+    description:
+      - A string containing the template contents.
+    type: str
+  state:
+    description:
+      - C(present) - state that is used to manage the template.
+      - C(absent) - delete the template.
+    choices: ["present", "absent"]
+    default: present
+    type: str
+
+notes:
+  - Supports C(check_mode).  Note that check mode always returns C(changed=true) for existing templates, even if the template would not actually change.
```

Comment:
@felixfontein hope a remark like this is fine, though I'm not sure about the usage of `C()` for `check_mode` here (a basic `grep` for `(check_mode)` in the repo had more usages of `C()` for that over `I()`...)

Reply:
@nilsding `I(...)` shouldn't be used for `check_mode`, since it is not a module/plugin option, but a task parameter, so it's correct :)